src/App.tsx

```diff
@@ -1,61 +1,148 @@
 import './App.scss';
+import { ChangeEvent, FC, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(id: number):User | null {
+  const foundUser = usersFromServer.find(user => user.id === id);
+
+  return foundUser || null;
+}
+
+function finderBiggestId(todos: Todo[]): number {
+  return Math.max(...todos.map(todo => todo.id));
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: FC = () => {
+  const [visibleTodos, setTodos] = useState(todos);
+  const [isSelectedUserError, setSelectionError] = useState(false);
+  const [isTitleError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const handlerTitleChange = (
+    event: ChangeEvent<HTMLInputElement>,
+  ) => {
+    const formattedTitle = event.target.value.replace(
+      /[^a-zA-Zа-яА-Я0-9\s]/g, '',
```

Comment:
Instead of using a regular expression to replace non-alphanumeric characters, consider using a validation message to inform the user about the allowed characters.